lib/adapters/http_server.js

```diff
@@ -48,40 +50,37 @@ var Read = source.extend({
                 // type is undefined
             }
 
-            Promise.try(function() {
-                if (req.method !== self.method) {
-                    throw self.runtime_error('RT-INVALID-HTTP-METHOD', {
+            Promise.try(() => {
+                if (req.method !== this.method) {
+                    throw this.runtimeError('RT-INVALID-HTTP-METHOD', {
                         types: 'POST'
                     });
                 }
 
                 if (format !== 'json' && format !== 'csv') {
-                    throw self.runtime_error('RT-INVALID-TYPE', {
+                    throw this.runtimeError('RT-INVALID-TYPE', {
                         types: 'csv, json'
                     });
                 }
 
 
-                var emit_points = [];
-
-                var parser =  parsers.getParser(format , {rootPath: self.rootPath});
-                return parser.parseStream(req, function(points) {
+                var parser = parsers.getParser(format , {rootPath: this.rootPath});
+                return parser.parseStream(req, (points) => {
                     if (!_.isArray(points)) {
                         points = [points];
                     }
 
-                    self.parseTime(points, self.timeField);
-                    emit_points = emit_points.concat(points);
+                    this.parseTime(points, this.timeField);
+                    this.points = this.points.concat(points);
                 })
-                .then(function() {
-                    self.emit(emit_points);
-
+                .then(() => {
```

Comment:
I didn't actually change this part of the code substantively, but yeah I can do that and I agree it would clean things up.